pointmatcher/DataPointsFilters/Geometry.h

```diff
@@ -0,0 +1,80 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the <organization> nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL ETH-ASL BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+#pragma once
+
+#include "PointMatcher.h"
+
+#include <vector>
+
+//! Surface normals estimation. Find the normal for every point using eigen-decomposition of neighbour points
+template<typename T>
+struct GeometryDataPointsFilter: public PointMatcher<T>::DataPointsFilter
+{
+	typedef PointMatcherSupport::Parametrizable Parametrizable;
+	typedef PointMatcherSupport::Parametrizable P;
+	typedef Parametrizable::Parameters Parameters;
+	typedef Parametrizable::ParameterDoc ParameterDoc;
+	typedef Parametrizable::ParametersDoc ParametersDoc;
+	typedef Parametrizable::InvalidParameter InvalidParameter;
+	
+	typedef typename PointMatcher<T>::Vector Vector;
+	typedef typename PointMatcher<T>::Matrix Matrix;	
+	typedef typename PointMatcher<T>::DataPoints DataPoints;
+	typedef typename PointMatcher<T>::DataPoints::InvalidField InvalidField;
+
+	inline static const std::string description()
+	{
+		return "This filter computes the level of ‘unstructureness’, ‘structureness’ and ’sphericality’ for each point based on the required eigen values.\n\n"
+		       "Required descriptors: eigValues (must be three).\n"
+		       "Produced descritors:  sphericality, unstructureness(optional), structureness(optional).\n"
+			   "Altered descriptors:  none.\n"
+			   "Altered features:     none.";
+	}
+	inline static const ParametersDoc availableParameters()
+	{
+		return {
+			{"keepUnstructureness", "whether the value of the unstructureness should be added to the pointcloud", "0"},
+			{"keepStructureness", "whether the value of the structureness should be added to the pointcloud", "0"},
+			};
+	}
+	
+	const bool keepUnstructureness;
+	const bool keepStructureness;
+
+    GeometryDataPointsFilter(const Parameters& params = Parameters());
+	virtual ~GeometryDataPointsFilter() {};
```

Comment:
Instead of an empty destructor can use the `default` keyword?

https://stackoverflow.com/questions/6502828/what-does-default-mean-after-a-class-function-declaration

Reply:
I think that both ways are clear to understand and the empty-body style is common to this library.

Reply:
Ok.